esindex/optimize.go

```diff
@@ -0,0 +1,27 @@
+package esindex
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Optimize the target index (or indices) to have the maximum number of
+// segments. Segments < 1 will default to 1.
+//
+// Optimize blocks until the operation completes.
+func Optimize(target string, segn int) error {
+	resp, err := http.Post(fmt.Sprintf("%s/_optimize?max_num_segments=%d", target, segn), "text/plain", nil)
```

Comment:
I have yet to be able to show a consistent improvement in bulk index throughput.... which goes against everything I've read and makes 0 intuitive sense... so hopefully I'm not just doing something wrong.

Internally we may want to experiment with much higher refresh intervals as it defaults to 1s which is far more often then we need. Anywhere from 10s to 10m seems like a better option for us, but that's a discussion for an internal ticket. :)